frontend/src/reducers/reducer.js

```diff
@@ -65,6 +53,22 @@ export const reducer = (state, action) => {
         case CLEAR_SEARCH:
             return { ...state, keyword: '' }
 
+        case USER_LOGGED_IN:
+            localStorage.setItem('TOKEN', payload);
+            return state
```

Comment:
Aren't the user details being stored when the user logs in?

Reply:
No user details are not stored when it logs in, whereas when we come to products page at that time based on userId user details are fetched from backend and stored in user state
Thanks